modules/db/src/blaze/db/node/resource_indexer.clj

```diff
@@ -84,7 +85,7 @@
   (sr/list-by-type search-param-registry (name (:fhir/type resource))))
 
 (defn- enhance-resource [last-updated resource]
-  (update resource :meta (fnil assoc #fhir/Meta{}) :lastUpdated last-updated))
+  (update resource :meta (fnil assoc (type/type #fhir/Meta{})) :lastUpdated last-updated))
```

Comment:
The code is right here. `#fhir/Meta{}` creates a new instance of the FHIR complex type `Meta`.